pt/lessons/specifics/debugging.md

```diff
@@ -1,23 +1,161 @@
+```
+
+Mas agora vamos para a parte empolgante - a depuração.
+
+Insira `require IEx: Iex.pry` depois da linha com `b = 0` e vamos tentar executar o código novamente.
+
+Você vai receber algo assim:
+
+```shell
+$ elixir test.exs
+warning: variable "b" is unused (if the variable is not meant to be used, prefix it with an underscore)
+  test.exs:2
+
+Cannot pry #PID<0.92.0> at TestMod.sum/1 (test.exs:5). Is an IEx shell running?
+34
+```
+
+Você deve perceber aquela mensagem crucial.
+
+Quando estiver executando uma aplicação, como sempre, o IEx gera essa mensagem em vez de bloquear a execução do programa.
+
+Para executar ela apropriadamente você precisa prefixar seu comando com `iex -S`.
+
+O que isso faz é executar `mix` dentro do comando `iex`, fazendo com que a aplicação seja executada em um modo especial, de tal jeito que chame `IEx.pry` para parar a execução da aplicação.
+
+Por exemplo, use `iex -S mix phx.serve` para depurar sua aplicação Phoenix. No nosso caso será `iex -r test.exs` para requerir o arquivo:
+
+```shell
+$ iex -r test.exs
+Erlang/OTP 21 [erts-10.3.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]
+
+warning: variable "b" is unused (if the variable is not meant to be used, prefix it with an underscore)
+  test.exs:2
+
+Request to pry #PID<0.107.0> at TestMod.sum/1 (test.exs:5)
+
+    3:     b = 0
+    4:
+    5:     require IEx; IEx.pry
+    6:
+    7:     a + b
+
+Allow? [Yn]
+```
+
+Depois de responder a confirmação com `y` ou pressionar Enter, você vai entrar no modo interativo.
+
+```shell
+ $ iex -r test.exs
+Erlang/OTP 21 [erts-10.3.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]
+
+warning: variable "b" is unused (if the variable is not meant to be used, prefix it with an underscore)
+  test.exs:2
+
+Request to pry #PID<0.107.0> at TestMod.sum/1 (test.exs:5)
+
+    3:     b = 0
+    4:
+    5:     require IEx; IEx.pry
+    6:
+    7:     a + b
+
+Allow? [Yn] y
+Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
+pry(1)> a
+34
+pry(2)> b
+0
+pry(3)> a + b
+34
+pry(4)> continue
+34
+
+Interactive Elixir (1.8.1) - press Ctrl+C to exit (type h() ENTER for help)
+iex(1)>
+BREAK: (a)bort (c)ontinue (p)roc info (i)nfo (l)oaded
+       (v)ersion (k)ill (D)b-tables (d)istribution
+```
+Para sair do IEx, você pode pressionar `Ctrl + C` duas vezes para sair do aplicativo ou digitar `continue` para seguir para o próximo breakpoint.
+
+Como você pode ver, é possivel executar qualquer código Elixir.
+
+Entretato, a limitação é que você não pode modificar nenhuma variável do código existente, devido a imutabilidade da linguagem.
+
+Apesar disso, você pode obter todos os valores das variáveis e executar qualquer computação.
+
+Nesse caso, o bug seria em `b` sendo reatribuído para 0, e a função `sum` sendo afetada como resultado.
+
+Claro, a linguagem já captorou esse bug até mesmo na primeira execução, mas isso é um exemplo!
+
+### IEx. Utilitários
```

Comment:
I think the `IEx.Helpers` name should be kept, since it refers to the [IEx.Helpers module](https://hexdocs.pm/iex/IEx.Helpers.html).